### ISSUE TITLE
The "user" option does not exist

### ISSUE THREAD

**Issue by shasta123** [nextcloud/sharelisting#310]
Hi,

I just noticed (after upgrading to NC 24) that occ sharing:list constantly outputs

The "user" option does not exist.

no matter what options (if any) I  specify.

On first glance I spot nothing in the log file.

Cheers,
Michael

**Comment by support-tt** [comment]
Same Problem here Nextcloud:
```
php occ status
  - installed: true
  - version: 24.0.0.12
  - versionstring: 24.0.0
  - edition: 
  - maintenance: false
  - needsDbUpgrade: false
  - productname: Nextcloud
  - extendedSupport: false
```
```
php occ sharing:list --user admin                        
The "user" option does not exist.  
sharing:list [-u|--user [USER]] [-p|--path [PATH]] [-t|--token [TOKEN]] [-f|--filter [FILTER]]
```
```
php occ sharing:list -u admin
The "user" option does not exist.  
sharing:list [-u|--user [USER]] [-p|--path [PATH]] [-t|--token [TOKEN]] [-f|--filter [FILTER]]

```

**Comment by th0ger** [comment]
Same issue here, first time user.
```console
$ php occ status
  - installed: true
  - version: 24.0.2.1
  - versionstring: 24.0.2
  - edition:
  - maintenance: false
  - needsDbUpgrade: false
  - productname: Nextcloud
  - extendedSupport: false
```

**Comment by Rdenney1967** [comment]
Same here
 - installed: true
  - version: 24.0.2.1
  - versionstring: 24.0.2
  - edition:
  - maintenance: false
  - needsDbUpgrade: false
  - productname: Nextcloud
  - extendedSupport: false

**Comment by CarlSchwan** [comment]
Fix incomming https://github.com/nextcloud/sharelisting/pull/312

**Comment by kroerig** [comment]
Are you shure, that this fixes the problem?
I change the function from public to proteced manually and got the following error:
`PHP Fatal error:  Access level to OCA\ShareListing\Command\ListShares::run() must be public (as in class OC\Core\Command\Base) in html/nextcloud/apps/sharelisting/lib/Command/ListShares.php on line 94`

**Comment by CarlSchwan** [comment]
> Are you shure, that this fixes the problem? I change the function from public to proteced manually and got the following error: `PHP Fatal error: Access level to OCA\ShareListing\Command\ListShares::run() must be public (as in class OC\Core\Command\Base) in html/nextcloud/apps/sharelisting/lib/Command/ListShares.php on line 94`

You need to also change run to execute and add `:int ` at the end

**Comment by kroerig** [comment]
Ok.
But now I get a new error:
```
An unhandled exception has been thrown:
TypeError: Return value of OCA\ShareListing\Command\ListShares::execute() must be of the type int, none returned in /html/nextcloud/apps/sharelisting/lib/Command/ListShares.php:123
Stack trace:
#0 /html/nextcloud/3rdparty/symfony/console/Command/Command.php(255): OCA\ShareListing\Command\ListShares->execute()
#1 /html/nextcloud/core/Command/Base.php(168): Symfony\Component\Console\Command\Command->run()
#2 /html/nextcloud/3rdparty/symfony/console/Application.php(1009): OC\Core\Command\Base->run()
#3 /html/nextcloud/3rdparty/symfony/console/Application.php(273): Symfony\Component\Console\Application->doRunCommand()
#4 /html/nextcloud/3rdparty/symfony/console/Application.php(149): Symfony\Component\Console\Application->doRun()
#5 /html/nextcloud/lib/private/Console/Application.php(211): Symfony\Component\Console\Application->run()
#6 /html/nextcloud/console.php(99): OC\Console\Application->run()
#7 /html/nextcloud/occ(11): require_once('/var/www/share....')
#8 {main}
```

**Comment by CarlSchwan** [comment]
` 		return 0;` at the end of the method :) See the changes in https://github.com/nextcloud/sharelisting/pull/312/files

**Comment by pReya** [comment]
@CarlSchwan I have the same problem. I just installed the app from the store (Nextcloud 23.0.6). Have the fixes in #312 not landed in a release, yet? Any way to run the version from master branch, if there's no release, yet?

**Comment by nursoda** [comment]
I confirm that 1.1.1 fixes the issue for me. Thanks!
